### PR TITLE
[SYCLomatic] Print some warnings when use --query-api-mapping.

### PR DESCRIPTION
**Pull request by tangjj11**
Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

**Review by intwanghao — APPROVED** [review]
LGTM

File: clang/lib/DPCT/Diagnostics.cpp [hunk 1]
```diff
@@ -48,7 +48,25 @@ std::unordered_map<int, DiagnosticsMessage> CommentIDTable;
 #undef DEF_WARNING
 #undef DEF_COMMENT
 
-std::unordered_set<int> APIQueryNeedReportWarningIDSet = {1086};
+std::unordered_set<int> APIQueryNeedReportWarningIDSet = {
+    1007, // API_NOT_MIGRATED
```
Comment (zhimingwang36) [inline clang/lib/DPCT/Diagnostics.cpp]: Let's remove 1007, it is confusing that dpct give out a mapping result, but at same time told 1007.

Comment (tangjj11) [reply clang/lib/DPCT/Diagnostics.cpp]: fixed

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**